stack/eslint-plugin-packlets/src/mechanics.ts

```diff
@@ -0,0 +1,122 @@
+      'packlet-importing-project-file':
+        'A local project file cannot be imported.' +
+        " A packlet's dependencies must be NPM packages and/or other packlets."
+    },
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false
+      }
+    ],
+    docs: {
+      description: '',
+      category: 'Best Practices',
+      recommended: 'warn',
+      url: 'https://www.npmjs.com/package/@rushstack/eslint-plugin-packlets'
+    }
+  },
+
+  create: (context: TSESLint.RuleContext<MessageIds, Options>) => {
+    // Example: /path/to/my-project/src/packlets/my-packlet/index.ts
+    const inputFilePath: string = context.getFilename();
+
+    // Example: /path/to/my-project/tsconfig.json
+    const tsconfigFilePath: string | undefined = ESLintUtils.getParserServices(
+      context
+    ).program.getCompilerOptions()['configFilePath'] as string;
+
+    const packletAnalyzer: PacketAnalyzer = PacketAnalyzer.analyzeInputFile(inputFilePath, tsconfigFilePath);
+    if (packletAnalyzer.nothingToDo) {
+      return {};
+    }
+
+    return {
+      // Match the first node in the source file.  Ideally we should be matching "Program > :first-child"
+      // so a warning doesn't highlight the whole file.  But that's blocked behind a bug in the query selector:
+      // https://github.com/estools/esquery/issues/114
+      Program: (node: TSESTree.Node): void => {
+        if (packletAnalyzer.error) {
+          context.report({
+            node: node,
+            messageId: packletAnalyzer.error.messageId,
+            data: packletAnalyzer.error.data
+          });
+        }
+      },
+
+      // ImportDeclaration matches these forms:
+      //   import { X } from '../../packlets/other-packlet';
+      //   import X from '../../packlets/other-packlet';
+      //   import type { X, Y } from '../../packlets/other-packlet';
+      //   import * as X from '../../packlets/other-packlet';
+      //
+      // ExportNamedDeclaration matches these forms:
+      //   export { X } from '../../packlets/other-packlet';
+      //
+      // ExportAllDeclaration matches these forms:
+      //   export * from '../../packlets/other-packlet';
+      //   export * as X from '../../packlets/other-packlet';
+      'ImportDeclaration, ExportNamedDeclaration, ExportAllDeclaration': (
+        node: TSESTree.ImportDeclaration | TSESTree.ExportNamedDeclaration | TSESTree.ExportAllDeclaration
+      ): void => {
+        if (node.source?.type === AST_NODE_TYPES.Literal) {
+          if (packletAnalyzer.projectUsesPacklets) {
+            // Extract the import/export module path
+            // Example: "../../packlets/other-packlet"
+            const modulePath = node.source.value;
+            if (typeof modulePath !== 'string') {
+              return;
+            }
+
+            if (!(modulePath.startsWith('.') || modulePath.startsWith('..'))) {
```

Comment:
Can we import from a file in the same folder like `import from 'filename.ts'` without having to do `./filename.ts`?

Reply:
We can't.  The module resolver would interpret `filename.ts` to be an NPM package with that name.